compiler/hash-typecheck/src/exhaustiveness/lower.rs

```diff
@@ -180,36 +180,43 @@ impl<'tc> LowerPatOps<'tc> {
             Pat::Constructor(ConstructorPat { args, .. }) => {
                 match reader.get_term(ty) {
                     Term::Level1(Level1Term::NominalDef(nominal_def)) => {
+                        // @@Todo: deal with variants
                         let (ctor, members) = match reader.get_nominal_def(nominal_def) {
                             NominalDef::Struct(struct_def) => match struct_def.fields {
                                 StructFields::Explicit(members) => {
-                                    (DeconstructedCtor::Single, members)
+                                    (DeconstructedCtor::Single, Some(members))
                                 }
                                 StructFields::Opaque => {
                                     panic!("got unexpected opaque struct-def here")
                                 }
                             },
+                            NominalDef::Unit(_) => (DeconstructedCtor::Single, None),
                             // @@EnumToUnion: when enums aren't a thing, do this with a union
                             NominalDef::Enum(_) => unreachable!(),
                         };
 
-                        // Lower the fields by resolving what positions the actual fields are
-                        // with the reference of the constructor's type...
-                        let fields = self.pat_lowerer().deconstruct_pat_fields(args, members);
+                        if let Some(members) = members {
```

Comment:
It would be nice if this was short-circuiting...
```suggestion
                        let members = Some(members) = members else {
                             return (ctor, vec![]);
                        }
```